ethcore/private-tx/src/encryptor.rs

```diff
@@ -187,9 +186,8 @@ impl SecretStoreEncryptor {
 
 	/// Try to unlock account using stored passwords
 	fn unlock_account(&self, account: &Address, accounts: Arc<AccountProvider>) -> bool {
-		let passwords = self.config.passwords.clone();
-		for password in passwords {
-			if let Ok(()) = accounts.unlock_account_temporarily(account.clone(), password) {
+		for password in &self.config.passwords {
```

Comment:
I don't get, why do we have one `account`, but multiple passwords? Also why not just pass password as additional argument to required method?

Reply:
User can use several different accounts: for signing transaction, for verification, for signing messages. Usage of password reworked